semgrep-core/src/tainting/Dataflow_tainting.ml

```diff
@@ -251,6 +251,56 @@ let union_taints_filtering_labels ~new_ curr =
           if req then Taints.add new_taint taints else taints)
     new_ curr
 
+let find_args_taints args_taints fdef =
+  let pos_args_taints, named_args_taints =
+    Common.partition_either Fun.id args_taints
+  in
+  let argpos_to_taints = Hashtbl.create 11 in
+  (* We first process the positional arguments, and then the named arguments.
+     This is OK because in Python, positional arguments must appear before named arguments.
+  *)
+  let remaining_params =
+    List.fold_left
+      (fun (i, params) taints ->
+        match params with
+        | [] ->
+            logger#error
+              "More args to function than there are positional arguments in \
+               function signature";
+            (i + 1, [])
+        | _ :: rest ->
+            Hashtbl.add argpos_to_taints (Left i) taints;
+            (i + 1, rest))
+      (0, fdef.G.fparams) pos_args_taints
+    |> snd
+  in
+  List.iter
+    (fun ((s, _), taints) ->
+      match
+        List.find_map
+          (function
+            | G.Param { pname = Some (s', _); _ } ->
+                if String.equal s s' then Some s' else None
+            | _ -> None)
+          remaining_params
+      with
+      | None -> logger#error "Cannot find named argument in function signature"
+      | Some s -> Hashtbl.add argpos_to_taints (Right s) taints)
+    named_args_taints;
+  fun (s, i) ->
+    let taint_opt =
+      match
+        ( Hashtbl.find_opt argpos_to_taints (Right s),
+          Hashtbl.find_opt argpos_to_taints (Left i) )
+      with
+      | Some taints, _ -> Some taints
+      | _, Some taints -> Some taints
```

Comment:
Can it be `Some _, Some _` ?

Reply:
Yes, but we would prefer that if an argument is identified as corresponding to a named parameter, then it should be associated with that parameter. The mapping from integer argument positions is only in the _absence_ of named arguments.